### PR TITLE
[feat] Visualize unrefed handles

### PR DESCRIPTION
**Pull request by Fishrock123**
Not without problems, but a good start.

See https://github.com/AndreasMadsen/dprof/issues/7

Looks like this:

screen shot 2016-08-29 at 6 59 47 pm

---

Note: the `nextTick` check isn't ideal but functions.

Better would be to track where a handle gets unrefed/re-refed/etc on the timeline via wrapping.

---

Note: After some investigation, I discovered that `TTY`(Wrap) and `Pipe`(Wrap) are always unrefed by some unknown mechanism. Even though `Pipe` actually has a `.ref()`, the following code shows that it does nothing whatsoever:

```
echo "console.log('hi'); process.stdout._handle.ref()" | node | grep ''
```

### CODE REVIEW

**Comment by AndreasMadsen** [comment]
Great. Only two things:

1) The root note is created with:

```js
const root = new Node('root', getCallSites(2));
```

this causes the name to be `String` now :) I would suggest we simply create a fake handle object:

```js
function RootWrap() {}
const root = new Node(new RootWrap(), getCallSites(2));
```

2) You should do the d3 part this way:

```js
  const barEnter = bar
    .enter().append('g')
      .attr('class', 'timeline')
      .classed('unrefed', (d) => d.unrefed);
```

and change the style to:

```css
  .timeline.unrefed .wait, .timeline.unrefed .callback, .timeline.unrefed .total {
    opacity: 0.5;
  }
```

It is more d3-ish, and the html/svg is less tied to the layout.

---

> Note: After some investigation, I discovered that TTY(Wrap) and Pipe(Wrap) are always unrefed by some unknown mechanism. Even though Pipe actually has a .ref(), the following code shows that it does nothing whatsoever:

haha, perhaps we could fix the `hasRef()` function in nodecore on those handle objects. But this is fine for now.

**Comment by Fishrock123** [comment]
Updated.

> haha, perhaps we could fix the `hasRef()` function in nodecore on those handle objects. But this is fine for now.

That's the plan, but so far I've had no luck determining how they even work today.

File: dprof.js [hunk 1]
```diff
@@ -100,7 +110,7 @@ asyncHook.addHooks({
   destroy: asyncDestroy
 });
 
-const root = new Node('root', getCallSites(2));
+const root = new Node({ constructtor: { name: 'root' } }, getCallSites(2));
```
Comment (AndreasMadsen) [inline dprof.js]: the spelling of `constructtor` is wrong.

**Comment by Fishrock123** [comment]
Whoops, rebased, all tests passing now.

**Comment by AndreasMadsen** [comment]
Great. 

> all tests passing now

Oh, you should add `unrefed` to the [`simplifyDump`](https://github.com/AndreasMadsen/dprof/blob/master/test/util.js#L29) method, then they won't parse, but just update the tests expectations. Also remember to update the format documentation to include `unrefed`. Besides that, it looks really good.

**Comment by AndreasMadsen** [comment]
Landed in 6b81b30b48a81b723275d40308d28b5ca4dd3b17. 996df59 caused some merge conflicts so I just dropped it and make one myself.

**Pull request closed without merging**